### PR TITLE
Migrate to Jakarta EE 9

### PR DESCRIPTION
**Pull request by hussainnm**

### CODE REVIEW

File: src/main/asciidoc/preface.adoc
Comment (ggam) [inline src/main/asciidoc/preface.adoc:128]: The tutorial is not bundled in GlassFish anymore so this reference should be removed.

**Comment by ggam** [comment]
Great work @hussainnm! Since we now have 2 PRs for the same purpose, but modiyfing different things, would you mind merging my branch into yours so I can close my PR? I think it will be easier for reviewers.

File: src/main/asciidoc/preface.adoc
Comment (hussainnm) [reply src/main/asciidoc/preface.adoc:128]: Could this be changed to "Represents the base installation directory for the Jakarta EE
Tutorial **after you download and extract it to the default location**".

Comment (hussainnm) [reply src/main/asciidoc/preface.adoc:128]: Or just "Represents the base installation directory for the Jakarta EE Tutorial" and third column as blank

**Comment by hussainnm** [comment]
> Great work @hussainnm! Since we now have 2 PRs for the same purpose, but modiyfing different things, would you mind merging my branch into yours so I can close my PR? I think it will be easier for reviewers.

Done. If you want to collaborate on this PR, you can raise PRs in my fork and I can merge it into this PR.

**Comment by ggam** [comment]
Thanks!!

**Comment by DmitriGit** [comment]
Hello Hussain, 

I'm Dmitri of the yersterday's email. I would want to contribute in creation of Jakarta EE Tutorials.

Thank you and have a nice day. :)
Dmitri.

**Comment by ggam** [comment]
@hussainnm impressive work. Really.

Just a question, what's the reasoning behind the 72 character limit? I guess the original files were length limited due to historical reasons, but nowadays all editors have word wrapping.

Also, the fixed size forces changes in multiple lines as soon as you add or remove a word (see https://github.com/eclipse-ee4j/jakartaee-tutorial/pull/239/commits/449dfe894e88c96a25dc53c950d7d23833c9330c#diff-0b377950484ffc2f08bfb2c67260289f0d6955472c3a5fd4f944f4c1dc8cbfc1L96 for an example).

**Comment by hussainnm** [comment]
The original document had this 72 character limit and after the Jakarta search and replace most of the paragraph were not falling in the 72 char limit. I wanted to maintain the same format as before. 

If we want to follow best practices for asciidoc, then each sentence can take one line and a new line between each paragraph.
By having one sentence per line, any changes made to the sentence will impact only that line and paragraph will be unaffected.

**Comment by ggam** [comment]
> The original document had this 72 character limit and after the Jakarta search and replace most of the paragraph were not falling in the 72 char limit. I wanted to maintain the same format as before.

Didn't notice that, thanks!

> If we want to follow best practices for asciidoc, then each sentence can take one line and a new line between each paragraph.
By having one sentence per line, any changes made to the sentence will impact only that line and paragraph will be unaffected.

That would be best IMO. But it's probably cleaner to do it as the final commit, after everything else is done, since it will move everything again.

**Comment by hussainnm** [comment]
I have now completed till Part 3. The following are the common issues I will be fixing in the remaining parts:

* line wrap at 72 characters
* remove section anchors
* Change link macro to xref
* Convert inline image to block image and add custom anchor
* Fix table format and add custom anchor
* Fix typos in source code like missing braces, closing tag, etc. 
* Fix numbered list (remove explicit numbers)

**Comment by hussainnm** [comment]
Dmitri has done a write up for [Jakarta EE 9 Highlights](https://github.com/hussainnm/jakartaee-tutorial/pull/1)

@ggam Please review and provide feedback before I merge it into this PR.

**Comment by ggam** [comment]
Looks good!

**Comment by keilw** [comment]
Beside details of labelling individual specs, before this can progress further, there must be a formal release tag for the old version Jakarta EE 8.

**Review by keilw — CHANGES_REQUESTED** [review]
8.1 (if that's the real version previously?) has to be released first with a formal release tag.

**Comment by hussainnm** [comment]
All chapters formatted.

Preview:
https://hussainnm.github.io/jakartaee-tutorial/

**Comment by ggam** [comment]
Congratulations and thanks @hussainnm. Now that the work is finished, could you also reformat to remove the 80-char limit we talked about earlier?

@keilw No version was ever released for the tutorial, among other things because it still contained a lot of Java EE references. Now that we have this PR and #246 in place, I wonder if we really should bother about the Jakarta EE 8 tutorial or simply aim to publish it for Jakarta EE 9 at the time 9.1 is released.

**Comment by arjantijms** [comment]
Amazing work! I spotted one "error", and that's that the term "Jakarta Server Faces" is still used. It's officially "Jakarta Faces". When we transitioned from Oracle to Eclipse a mistake was made, and it was indeed erroneously called "Jakarta Server Faces". This mistakte has finally been corrected a short while ago. 

See also: https://projects.eclipse.org/projects/ee4j.faces

**Comment by keilw** [comment]
@arjantijms / all, do you know, where is the release tag for the Jakarta EE 8 version?
There was an "initial import" 2 years ago but never a Jakarta EE 8 tag. 
Is that also still in progress and this would be the next step or why is there no Jakarta EE 8 compliant tag, be it called **8.0**, **8.1** or whatever?

**Comment by ggam** [comment]
There was no tutorial release for Jakarta EE 8. Until now we lacked updated
images (the existing ones referred to Java EE) and committer reviews on
this repository are rare, so we basically continued working on it.

Jakarta EE 8 is identical yo Java EE 8 so I would just skip that version
and publish this EE 9 version as the first one. Hopefully we'll get more
traction once any version is available online.

**Comment by arjantijms** [comment]
> Hopefully we'll get more
> traction once any version is available online.

Hope so too. The Jakarta EE 9.1 release is supposed to be around next week, so it would be really great to release the tutorial at the same time ;)

**Comment by keilw** [comment]
Ok so that means the "initial import" tag should work (mostly without major changes) for Jakarta EE 8?

**Comment by ggam** [comment]
Current master versión should be fine, with most Java EE references already
changed to Jakarta EE.

**Comment by hussainnm** [comment]
> Amazing work! I spotted one "error", and that's that the term "Jakarta Server Faces" is still used. It's officially "Jakarta Faces". When we transitioned from Oracle to Eclipse a mistake was made, and it was indeed erroneously called "Jakarta Server Faces". This mistakte has finally been corrected a short while ago.
> 
> See also: https://projects.eclipse.org/projects/ee4j.faces

Done.

**Comment by arjantijms** [comment]
@hussainnm thanks a lot!

**Comment by hussainnm** [comment]
> Congratulations and thanks @hussainnm. Now that the work is finished, could you also reformat to remove the 80-char limit we talked about earlier?

@ggam I have started working on the above. I am doing it in a separate branch on my fork. You can follow my progress in the below PR.

https://github.com/hussainnm/jakartaee-tutorial/pull/4

**Comment by ggam** [comment]
Fantastic. Just curious, do you plan to leave this PR as draft until the other one is ready?

It's not reasonable to expect anyone will review the content of this modification. In the past, some other massive PRs of mine have been merged as-is. I hope the same can be done here: accept the PR and open new issues as people use the tutorial and find problems.

Once you're done we should ask for someone with committer rights to do a high level review and approve it.

**Comment by hussainnm** [comment]
I would say this PR is almost ready for review. I still see 3 images need to be converted to SVG. So lets keep this in draft till Dmitri finishes up.

Regarding waiting for reflow PR to complete, I have just started it and expect to complete it in 15-20 days (3-4 chapters in a day).

**Comment by ggam** [comment]
Ok, let's wait then for @DmitriGit and request review for this PR after that :+1:.

**Comment by DmitriGit** [comment]
A moment ago committed the last image (Figure 62-2). Now I review all images for quantity comparing, but in my opinion all tutorial images are done and delivered at this point. 

    On Sunday, April 11, 2021, 06:09:39 PM GMT+2, Guillermo González de Agüero ***@***.***> wrote:  
 
 


Ok, let's wait then for @DmitriGit and request review for this PR after that 👍 .

**Comment by DmitriGit** [comment]
Hi Team,
4 images not converted by me in SVG are the images that are not present in a new version of the tutorial. And in particular:     1) jakartaeett_dt_006.png     2) jakartaeett_dt_007.png     3) jakartaeett_dt_008.png     4) jakartaeett_dt_009.pngare the images of old "Java EE 7 APIs"'a paragraph. If you want i can convert them.
Actually I have now a doubt about "jakartaeett_dt_067_slctn.png" and "jakartaeett_dt_017.png" - give me some hours for verify and commit the right version.
Thank you!Dmitri.

    On Sunday, April 11, 2021, 06:08:16 PM GMT+2, Hussain ***@***.***> wrote:  
 
 


I would say this PR is almost ready for review. I still see 3 images need to be converted to SVG. So lets keep this in draft till Dmitri finishes up.

Regarding waiting for reflow PR to complete, I have just started it and expect to complete it in 15-20 days (3-4 chapters in a day).

**Comment by hussainnm** [comment]
> Hi Team, 4 images not converted by me in SVG are the images that are not present in a new version of the tutorial. And in particular:     1) jakartaeett_dt_006.png     2) jakartaeett_dt_007.png     3) jakartaeett_dt_008.png     4) jakartaeett_dt_009.pngare the images of old "Java EE 7 APIs"'a paragraph. If you want i can convert them. 

@DmitriGit I think it would be great if you can convert these images. The current images highlight what was added newly in Java EE 7. We can add back the section and showcase what got added and removed in Jakarta EE 9. The images can evolve in future versions.

**Comment by DmitriGit** [comment]
Ok, no problem - I'll do it quickly. 
One question: on these old images I leave reference to Java EE 7, right? If necessary I can create two versions - one for Java EE 7 and another one for... Jakarta of some version if you want...


    On Tuesday, April 13, 2021, 03:46:50 AM GMT+2, Hussain ***@***.***> wrote:  
 
 



Hi Team, 4 images not converted by me in SVG are the images that are not present in a new version of the tutorial. And in particular:     1) jakartaeett_dt_006.png     2) jakartaeett_dt_007.png     3) jakartaeett_dt_008.png     4) jakartaeett_dt_009.pngare the images of old "Java EE 7 APIs"'a paragraph. If you want i can convert them.


@DmitriGit I think it would be great if you can convert these images. The current images highlight what was added newly in Java EE 7. We can add back the section and showcase what got added and removed in Jakarta EE 9. The images can evolve in future versions.

**Comment by hussainnm** [comment]
Create only one version with Jakarta EE names.

**Comment by DmitriGit** [comment]
Done. 4 images of old "Java EE 7 API" section are created and committed.
Now I start to verify all 68 images to the possible errata presence of old "Java EE" terminology and to correct some Figures with a standard elements (for example, if you view images committed these days you can note that Database element is nicer and more standard than created by me previously).
I plan to finish this step until Friday 16 April, ok?


    On Tuesday, April 13, 2021, 08:56:17 AM GMT+2, Hussain ***@***.***> wrote:  
 
 


Create only one version with Jakarta EE names.

**Comment by DmitriGit** [comment]
Dmitri wrote:       Actually I have now a doubt about "jakartaeett_dt_067_slctn.png" and "jakartaeett_dt_017.png" - give me some hours for verify and          commit the right version.

Done - corrected and pushed.
Dmitri.

    On Monday, April 12, 2021, 08:15:14 PM GMT+2, Dmitri Cerkas ***@***.***> wrote:  
 
  Hi Team,
4 images not converted by me in SVG are the images that are not present in a new version of the tutorial. And in particular:     1) jakartaeett_dt_006.png     2) jakartaeett_dt_007.png     3) jakartaeett_dt_008.png     4) jakartaeett_dt_009.pngare the images of old "Java EE 7 APIs"'a paragraph. If you want i can convert them.
Actually I have now a doubt about "jakartaeett_dt_067_slctn.png" and "jakartaeett_dt_017.png" - give me some hours for verify and commit the right version.
Thank you!Dmitri.

    On Sunday, April 11, 2021, 06:08:16 PM GMT+2, Hussain ***@***.***> wrote:  
 
 


I would say this PR is almost ready for review. I still see 3 images need to be converted to SVG. So lets keep this in draft till Dmitri finishes up.

Regarding waiting for reflow PR to complete, I have just started it and expect to complete it in 15-20 days (3-4 chapters in a day).

**Comment by hussainnm** [comment]
@DmitriGit I see your commit message mentions image for 16-1, but jakartaeett_dt_017.svg is not checked-in.

**Comment by DmitriGit** [comment]
You are aaaabsolutely right, Hussain, thank you! Corrected and pushed.
Do you find strange so much difference between the image file names (for example, "jakartaeett_dt_057") and image number in the tutorial ("Figure_56_4" that corresponds to "jakartaeett_dt_057") ? This creates confusion... May be one day we can aligh two types of names, what do you think about it?
Thank you!
    On Thursday, April 15, 2021, 09:03:23 AM GMT+2, Hussain ***@***.***> wrote:  
 
 


@DmitriGit I see your commit message mentions image for 16-1, but jakartaeett_dt_017.svg is not checked-in.

**Comment by hussainnm** [comment]
The image file names and figure number will go out of sync as soon as new sections are added in between or removed. I think it is best we add new images with incrementing file name and let asciidoctor assign the figure number.

**Comment by DmitriGit** [comment]
Ok, you are right, it's logical.

**Comment by DmitriGit** [comment]
Hi Hussain and Guillermo,
firstly - I'm free and if you have to give me something to do I'll be glad! :)
Secondly - as said time ago - I have a suggestions on the content of the Tutorial (nothing that would make much impact - just a few remarks on the content of a some paragraphs here and there). If, according to your opinion, my suggestions will make sense, we will obtain even more comprehensive and even more logical tutorial.Can we discuss these changes in this thread or you prefer create a separate one?Promise to be very clear in the explanations to steal as little time as possible from you.
Dmitri.
    On Tuesday, April 13, 2021, 03:46:50 AM GMT+2, Hussain ***@***.***> wrote:  
 
 



Hi Team, 4 images not converted by me in SVG are the images that are not present in a new version of the tutorial. And in particular:     1) jakartaeett_dt_006.png     2) jakartaeett_dt_007.png     3) jakartaeett_dt_008.png     4) jakartaeett_dt_009.pngare the images of old "Java EE 7 APIs"'a paragraph. If you want i can convert them.


@DmitriGit I think it would be great if you can convert these images. The current images highlight what was added newly in Java EE 7. We can add back the section and showcase what got added and removed in Jakarta EE 9. The images can evolve in future versions.

**Comment by hussainnm** [comment]
It will be preferable if you could create an issue with your suggestions/improvements and we can discuss over there.

File: src/main/asciidoc/preface.adoc
Comment (hussainnm) [reply src/main/asciidoc/preface.adoc]: I have updated this to user's home directory.

**Comment by hussainnm** [comment]
I have updated the document to follow one line per sentence style.

I am marking this PR as "Ready for Review"

**Comment by DmitriGit** [comment]
When I do a search within the manual _(https://hussainnm.github.io/jakartaee-tutorial/)_ with a browser search (Ctrl-F), I still find references to "Java EE". For example:
1) "Java EE application client" in "Creating Shared Subscriptions" paragraph _("https://hussainnm.github.io/jakartaee-tutorial/#creating-shared-subscriptions")_,
2) "Java EE components" in "Overview of Using Jakarta Messaging" paragraph _(https://hussainnm.github.io/jakartaee-tutorial/#overview-of-using-jakarta-messaging)_,

and others ...

**Review by keilw — CHANGES_REQUESTED** [review]
This needs to wait till the 8.1 release tag is fixed and properly done.

**Comment by hussainnm** [comment]
> When I do a search within the manual _(https://hussainnm.github.io/jakartaee-tutorial/)_ with a browser search (Ctrl-F), I still find references to "Java EE". For example:
> 
> 1. "Java EE application client" in "Creating Shared Subscriptions" paragraph _("https://hussainnm.github.io/jakartaee-tutorial/#creating-shared-subscriptions")_,
> 2. "Java EE components" in "Overview of Using Jakarta Messaging" paragraph _(https://hussainnm.github.io/jakartaee-tutorial/#overview-of-using-jakarta-messaging)_,
> 
> and others ...

Fixed. Only Java EE 5 references remain which are valid.

**Comment by hussainnm** [comment]
Merge conflicts are resolved.

**Comment by arjantijms** [comment]
> Merge conflicts are resolved.

Thanks a bunch! Merging now.

**Comment by DmitriGit** [comment]
May be it was done on purpose, but number of every part of the Tutorial and a row describing the content of the part have different numbering. For example:
"
_**2**. Introduction
Part **I** introduces the platform, the tutorial, and the examples._
"
in this case:
1) Part "Introduction" has a number "2", but
2) row describing the content of the part have a number "1 (roman I)"

the same is for remaining content of the tutorial, for example:
"
_**4**. The Web Tier
Part **III** explores the technologies in the web tier._
"
(Part "4", but "Part III")

**Comment by hussainnm** [comment]
@DmitriGit The github pages for jakartaee-tutorial looks fine to me.
https://eclipse-ee4j.github.io/jakartaee-tutorial/

If you are looking at github pages for my fork. I have updated it now to latest master. The part and chapter numbers should align.

**Pull request merged**
